src/neuralnet/input_layer/store.cc

```diff
@@ -70,30 +76,44 @@ void StoreInputLayer::ComputeFeature(int flag,
     }
   }
   for (int k = 0; k < batchsize_; k++) {
-    if (!store_->Read(&key, &val)) {
+    if (!store_->Read(&buf_keys_[k], &buf_vals_[k])) {
       store_->SeekToFirst();
-      CHECK(store_->Read(&key, &val));
+      CHECK(store_->Read(&buf_keys_[k], &buf_vals_[k]));
+    }
+  }
+}
+
+void StoreInputLayer::ComputeFeature(int flag,
+    const vector<Layer*>& srclayers) {
+  // if prefetching, wait for the thread to finish
+  if (layer_conf_.store_conf().prefetching()) {
+    if (thread_ == nullptr) {
+      buf_keys_.resize(batchsize_);
```

Comment:
buf_keys and buf_vals must be allocated no matter prefetching is enabled or not, because they are used by fetch_data().
make test failed...
It works if prefetching's default value is false.
I haven't found the bug..